<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>784</width>
    <height>526</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: rgb(0, 85, 127)</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_4">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_5" stretch="10,0">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="0,10,0,10,0">
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_4" stretch="6,6">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <spacer name="verticalSpacer_2">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QLabel" name="chooseLabel">
              <property name="layoutDirection">
               <enum>Qt::RightToLeft</enum>
              </property>
              <property name="styleSheet">
               <string notr="true">font-size: 16px; color: rgb(255, 255, 255); font-weight: bold;</string>
              </property>
              <property name="text">
               <string>Choose limit number</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="limitSpinBox">
              <property name="styleSheet">
               <string notr="true">QSpinBox
{
 background-color: #fff;
 border: 1px solid #ccc;
 color: rgb(0, 0, 0)
 border-radius: 5px;
 padding: 5px;
 font-size: 18px;
}
QSpinBox::up-button, QSpinBox::down-button {
    background-color: #555;
    border: 1px solid #777;
    width: 16px;
    subcontrol-origin: border;
    subcontrol-position: top right; 
}

QSpinBox::down-button {
    subcontrol-position: bottom right;
}

QSpinBox::up-button:hover, QSpinBox::down-button:hover {
    background-color: #777;
}

QSpinBox::up-arrow, QSpinBox::down-arrow {
    width: 10px;
    height: 10px;
    image: url(:/icons/up-arrow.png); 
}

QSpinBox::down-arrow {
    image: url(:/icons/down-arrow.png);
}</string>
              </property>
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>1000000</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <widget class="QLabel" name="sliderLabel">
              <property name="layoutDirection">
               <enum>Qt::RightToLeft</enum>
              </property>
              <property name="styleSheet">
               <string notr="true">font-size: 16px; color: rgb(255, 255, 255); font-weight: bold;</string>
              </property>
              <property name="text">
               <string>Choose number of threads</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSlider" name="threadCountSlider">
              <property name="styleSheet">
               <string>
           QSlider { background: #e0e0e0; height: 10px; border-radius: 5px; }
           QSlider::handle:horizontal { background: #4CAF50; border-radius: 50%; width: 20px; height: 20px; margin-top: -5px; }
           QSlider::handle:horizontal:hover { background: #45a049; }
           QSlider::handle:horizontal:pressed { background: #388e3c; }
         </string>
              </property>
              <property name="value">
               <number>1</number>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_2">
              <item>
               <spacer name="horizontalSpacer_4">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QLabel" name="numberOfThreadsLabel">
                <property name="styleSheet">
                 <string notr="true">QLabel
{
font-size:18px;
color: rgb(31, 31, 31);
font-weight: bold;
}</string>
                </property>
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer_5">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </item>
            <item>
             <spacer name="verticalSpacer_5">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QLabel" name="outputTextLabel">
            <property name="layoutDirection">
             <enum>Qt::RightToLeft</enum>
            </property>
            <property name="styleSheet">
             <string notr="true">font-size: 16px; color: rgb(255, 255, 255); font-weight: bold;</string>
            </property>
            <property name="text">
             <string>Information</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QTextEdit" name="outputText">
            <property name="styleSheet">
             <string>background-color: #f0f0f0; border: 1px solid #ccc; border-radius: 5px; padding: 10px; font-size: 14px; color: #333;</string>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_4">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <widget class="QPushButton" name="start_btn">
              <property name="styleSheet">
               <string notr="true">QPushButton {
background-color: #4CAF50; 
color: rgb(31, 31, 31);
font-weight: bold;
border: 1px solid #4CAF50; 
padding: 10px 20px;
 border-radius: 5px; 
font-size: 18px;  }
QPushButton::hover {
    background-color: rgb(0, 85, 0);
    border-color: rgb(0, 85, 0);
}
QPushButton::disabled
{
    background-color: #C7DEB9;
    border-color: #C7DEB9;
}</string>
              </property>
              <property name="text">
               <string>Start</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="stop_btn">
              <property name="styleSheet">
               <string notr="true">QPushButton 
{background-color: #f44336;
 color: rgb(31, 31, 31);
 border: #f44336;
 padding: 10px 20px; 
 border-radius: 5px; 
 font-size: 18px; 
font-weight:bold;
}
QPushButton::hover {
    background-color: rgb(85, 0, 0);
    border-color: rgb(85, 0, 0);
}
QPushButton::disabled
{
    background-color: #DECBCB;
    border-color: #DECBCB;
}
</string>
              </property>
              <property name="text">
               <string>Stop</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="exit_btn">
              <property name="styleSheet">
               <string notr="true">QPushButton{
background-color: #9e9e9e; 
color: rgb(31, 31, 31);
border: 1px solid #9e9e9e; 
padding: 10px 20px; 
border-radius: 5px; 
font-size: 18px;
font-weight:bold;
}
QPushButton::hover
{
background-color: rgb(43, 37, 33);
border: 1px solid rgb(43, 37, 33);
}
</string>
              </property>
              <property name="text">
               <string>Exit</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="horizontalSpacer_3">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="verticalSpacer_3">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>784</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <property name="separatorsCollapsible">
     <bool>false</bool>
    </property>
    <property name="toolTipsVisible">
     <bool>false</bool>
    </property>
    <addaction name="actionLanguage"/>
   </widget>
   <addaction name="menuEdit"/>
  </widget>
  <action name="actionLanguage">
   <property name="text">
    <string>Language</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
